src/sst.rs

```diff
@@ -78,9 +78,17 @@ impl SsTableFormat {
 
     pub(crate) fn decode_filter(
         &self,
-        filter_bytes: Bytes,
+        bytes: Bytes,
         compression_codec: Option<CompressionCodec>,
     ) -> Result<BloomFilter, SlateDBError> {
+        let checksum_sz = std::mem::size_of::<u32>();
```

Comment:
Now that we have the decode checksum logic in 3 places (decode_filter, decode_index, decode_block), could you place this logic in a separate function that can be called from all three spots?

Reply:
Oh yeah, that sounds much better, will do.